src/main/java/io/airlift/units/Count.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.units;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static io.airlift.units.Preconditions.checkArgument;
+import static java.lang.Math.floor;
+import static java.util.Objects.requireNonNull;
+
+public class Count
+        implements Comparable<Count>
+{
+    private static final Pattern PATTERN = Pattern.compile("^\\s*(\\d+(?:\\.\\d+)?)\\s*([a-zA-Z]?)\\s*$");
+
+    // We iterate over the COUNT_UNITS constant in convertToMostSuccinctDataSize()
+    // instead of Unit.values() as the latter results in non-trivial amount of memory
+    // allocation when that method is called in a tight loop. The reason is that the values()
+    // call allocates a new array at each call.
+    private static final Unit[] COUNT_UNITS = Unit.values();
+
+    public static Count succinct(long bytes)
+    {
+        return succinctCount(bytes, Unit.SINGLE);
+    }
+
+    public static Count succinctCount(double size, Unit unit)
+    {
+        return new Count(size, unit).convertToMostSuccinctCount();
+    }
+
+    private final double value;
```

Comment:
Do we want an inexact value for this?  IIRC, a single `long` would get us to 4000 quadrillion.

Reply:
I was rather thinking about case where something could have fraction part (like water or mass). Though I do not have strong opinion. Would you like me to change this?

Reply:
I'm not sure.  I can't think of any case where I would want to use this for non-integral counts.  For water or mass, I assume we would introduce `Volume` and `Mass` units.  Maybe @electrum or @martint  have thoughts.

Reply:
Notice that `DataSize` is also using `double`. Using `double` here is useful when doing a coversions between different units, like from KB to MB (as fractional part appears). The same happens for `Count`. When `long` is used then `Count` become precise more precise, but not all conversion are possible. 

See #9.

Reply:
I renamed this class to `Quantity` so it now it clear that it may contain fractional part. Also now, it is a bit more general.